modules/oe_translation_remote/tests/src/Functional/RemoteTranslationTest.php

```diff
@@ -36,4 +37,66 @@ public function testRemoteTranslationRoute(): void {
+    $this->assertSession()->pageTextContains('Remote Translator Provider entities');
+    $this->assertSession()->linkExists('Add Remote Translator Provider');
+    $this->clickLink('Add Remote Translator Provider');
+
+    // Assert form elements.
+    $this->assertSession()->pageTextContains('Add remote translator provider');
+    $this->assertSession()->fieldExists('Name');
+    $this->assertSession()->pageTextContains('Name of the Remote translation provider.');
+    $this->assertSession()->elementAttributeContains('css', 'input#edit-label', 'required', 'required');
+    $select = $this->assertSession()->selectExists('Plugin');
+    // Assert that all the installed plugins are available in the select.
+    $options = [];
+    foreach ($select->findAll('xpath', '//option') as $element) {
+      $options[$element->getValue()] = trim($element->getText());
+    }
+    $expected_options = [
+      'Please select a plugin',
+      'Remote one',
+      'Remote two',
+    ];
+    $this->assertEqualsCanonicalizing($expected_options, $options);
+    $this->assertSession()->pageTextContains('The plugin to be used with this translator.');
+    $this->assertSession()->elementAttributeContains('css', 'select#edit-plugin', 'required', 'required');
+
+    // Assert different configuration form based on selected plugin.
+    $this->getSession()->getPage()->selectFieldOption('Plugin', 'Remote one');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->assertSession()->pageTextContains('Plugin configuration for Remote one');
+    $this->assertSession()->pageTextContains('This plugin does not have any configuration options.');
+    $this->getSession()->getPage()->selectFieldOption('Plugin', 'Remote two');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->assertSession()->pageTextContains('Plugin configuration for Remote two');
+    $this->assertSession()->fieldExists('A test configuration string');
+
+    // Create a new remote translator provider.
+    $this->getSession()->getPage()->fillField('Name', 'Test provider');
+    $this->getSession()->getPage()->fillField('A test configuration string', 'Plugin configuration.');
+    $this->getSession()->getPage()->pressButton('Save');
+    $this->getSession()->getPage()->fillField('Machine-readable name', 'test_provider');
+    $this->getSession()->getPage()->pressButton('Save');
+    $this->assertSession()->pageTextContains('Created the Test provider Remote Translator Provider.');
+
+    // Edit the provider.
+    $this->clickLink('Edit');
+    $this->assertSession()->fieldValueEquals('Name', 'Test provider');
+    $this->assertSession()->fieldValueEquals('A test configuration string', 'Plugin configuration.');
+    $this->getSession()->getPage()->fillField('Name', 'Test provider edited');
+    $this->getSession()->getPage()->pressButton('Save');
+    $this->assertSession()->pageTextContains('Saved the Test provider edited Remote Translator Provider.');
```

Comment:
Do also a load of the config entity and assert all the exported values are in the correct format.